packages/shared/lib/core/layer-2/types/abi.type.ts

```diff
@@ -0,0 +1 @@
+export type AbiType = 'function' | 'constructor' | 'event' | 'fallback'
```

Comment:
Perhaps we should use an `enum` if possible, imo they're slightly more friendly to use than string union types.

Reply:
updated ✅